### DISCUSSION TITLE
C# await hangs when executed from pulumi but not from standalone program

### DISCUSSION THREAD

**Discussion by rgl** [pulumi/pulumi#10674] [General]
Can you help me understand why the following `await` call never returns when called from within a Pulumi program but works when called from a stand-alone program?

```csharp

return await Deployment.RunAsync(async () =>
{
    var baseVolumeName = await GetBaseVolumeName();
...
});
```

The full example at https://github.com/rgl/dotnet-pulumi-libvirt-ubuntu-example/blob/use-latest-ubuntu-volume/Program.cs#L17

The same call made from [a standalone program](https://github.com/rgl/dotnet-cli-wrap-virsh-vol-list) works as expected, so I'm not sure of what is going on.

Can you please help?

Please find the logs at [pulumi.log](https://github.com/pulumi/pulumi/files/9540042/pulumi.log) (generated with `TF_LOG=TRACE pulumi up --logtostderr --logflow -v=9 2>pulumi.log`).

**Comment by nimbinatus** [comment]
Hi there! This seems like something that's worth passing on as a bug. I'll open an issue from this discussion and leave a link here. The engineering team will take a look!

**Reply by nimbinatus** [reply]
Issue opened: https://github.com/pulumi/pulumi/issues/10712
